src/java.base/share/classes/jdk/internal/vm/vector/VectorSupport.java

```diff
@@ -294,42 +344,221 @@ public static long multiFieldOffset(Class<? extends VectorPayloadMF> cls) {
+    public primitive static class VectorPayloadMF512B extends VectorPayloadMF {
         @MultiField(value = 64)
         byte mfield = 0;
-        static long MFOFFSET = multiFieldOffset(VectorPayloadMF512.class);
+        static long MFOFFSET = multiFieldOffset(VectorPayloadMF512B.class);
+
+        @Override
+        public long multiFieldOffset() { return MFOFFSET;}
+    }
+
+    public primitive static class VectorPayloadMF64S extends VectorPayloadMF {
+        @MultiField(value = 4)
+        short mfield = 0;
+        static long MFOFFSET = multiFieldOffset(VectorPayloadMF64S.class);
+
+        @Override
+        public long multiFieldOffset() { return MFOFFSET;}
+    }
+
+    public primitive static class VectorPayloadMF128S extends VectorPayloadMF {
+        @MultiField(value = 8)
+        short mfield = 0;
+        static long MFOFFSET = multiFieldOffset(VectorPayloadMF128S.class);
+
+        @Override
+        public long multiFieldOffset() { return MFOFFSET;}
+    }
+
+    public primitive static class VectorPayloadMF256S extends VectorPayloadMF {
+        @MultiField(value = 16)
+        short mfield = 0;
+        static long MFOFFSET = multiFieldOffset(VectorPayloadMF256S.class);
+
+        @Override
+        public long multiFieldOffset() { return MFOFFSET;}
+    }
+
+    public primitive static class VectorPayloadMF512S extends VectorPayloadMF {
+        @MultiField(value = 32)
+        short mfield = 0;
+        static long MFOFFSET = multiFieldOffset(VectorPayloadMF512S.class);
+
+        @Override
+        public long multiFieldOffset() { return MFOFFSET;}
+    }
+
+    public primitive static class VectorPayloadMF64I extends VectorPayloadMF {
+        @MultiField(value = 2)
+        int mfield = 0;
+        static long MFOFFSET = multiFieldOffset(VectorPayloadMF64I.class);
+
+        @Override
+        public long multiFieldOffset() { return MFOFFSET;}
+    }
+
+    public primitive static class VectorPayloadMF128I extends VectorPayloadMF {
+        @MultiField(value = 4)
+        int mfield = 0;
+        static long MFOFFSET = multiFieldOffset(VectorPayloadMF128I.class);
+
+        @Override
+        public long multiFieldOffset() { return MFOFFSET;}
+    }
+
+    public primitive static class VectorPayloadMF256I extends VectorPayloadMF {
+        @MultiField(value = 8)
+        int mfield = 0;
+        static long MFOFFSET = multiFieldOffset(VectorPayloadMF256I.class);
+
+        @Override
+        public long multiFieldOffset() { return MFOFFSET;}
+    }
+
+    public primitive static class VectorPayloadMF512I extends VectorPayloadMF {
+        @MultiField(value = 16)
+        int mfield = 0;
+        static long MFOFFSET = multiFieldOffset(VectorPayloadMF512I.class);
+
+        @Override
+        public long multiFieldOffset() { return MFOFFSET;}
+    }
+
+    public primitive static class VectorPayloadMF64L extends VectorPayloadMF {
+        @MultiField(value = 1)
+        long mfield = 0;
```

Comment:
Can remove the MultiField annotation?

Reply:
Annotations are propagated to both oops and ci model, and we do special handlings for multifields.

Reply:
Yes, so besides the vectorization, any other special handling of the multifield? I think vectorization is not affected if it is moved for one element count. Anyway, adding this annotation is also fine to me. Thanks!